Source/WTF/wtf/PlatformHave.h

```diff
@@ -278,7 +278,7 @@
 #define HAVE_FLOAT16_INSTRUCTION 1
 #endif
 
-#if ((CPU(ARM64) || (CPU(X86_64) && !PLATFORM(PLAYSTATION))) && defined(__FLT16_MANT_DIG__))
+#if !OS(LINUX) && !PLATFORM(PLAYSTATION) && (CPU(ARM64) || CPU(X86_64))
```

Comment:
I would recommend doing the opposite: use the compiler test everywhere except Cocoa platforms.

Notably, whether `_Float16` is available is really complicated. Look at https://github.com/simd-everywhere/simde/pull/1182 and notice that it depends not just on architecture, but also on compiler, compiler version, and programming language. (E.g. on aarch64 it's only available in C++ since GCC 13, or in C since GCC 7, or in Clang since Clang 15). So this condition doesn't come close to detecting it properly. But I'm not sure whether it actually matters whether we are Linux or PlayStation or FreeBSD or Fuschia....

Reply:
Thank you Michael, very interesting. I reworked the patch according to your comment.